design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,582 @@
+      
+      3.3.1 **Key Adapter** - Responsible of mapping source key to hist corresponding Destination key value 
+    
+    * GCP implementation - use a key adapter to find the corresponding key to validate
+    * Azure needs refactor - same Azure xms_mirid logic
+    * K8s needs refactor - same K8S logic
+    * JWT design phase - straightforward validation 
+
+#### GCP authenticator flow
+![Authn gcp flow](authn-gcp-flow.png)
+
+
+### Backwards compatibility
+None because this is a new feature.
+
+### Performance
+GCP authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+- Conjur
+
+- DAP
+
+## Security
+* **Future support in Service account key needs to be done very carefully** since there is a concern of impersonate other Conjur hosts, 
+because the Service account key enable you to control the content of the JWT token. 
+The reason it is not a concern in our use case, because we are validating only tokens which supplied by google metadata server, 
+and signed by google private keys which are not expose to the apps in contrast to the Service account key.
+
+## Test Plan
+[//]: # (Fill in the table below to depict the tests that should run to validate your solution)
+[//]: # (You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#)
+
+| **Title** | **Given** | **When** | **Then** | **Comment** |
+|-----------|-----------|----------|----------|-------------|
+|           |           |          |          |             |
+|           |           |          |          |             |
+
+## Logs
+
+### Error Log level
+
+|    | Scenario                                                                | Log message                                                                            | Comment            |
+|--- |-----------------------------------------------------------------------  |----------------------------------------------------------------------------------------|------------------------|
+| 1  | Authenticator is not enabled (in DB/ENV)                                | Authenticator '{0-authenticator-name}' is not enabled                                  | |
+| 2  | Webservice is not defined in a Conjur policy                            | Webservice '{0-webservice-name}' wasn't found                                          | |
+| 3  | Host is not permitted to authenticate with the webservice               | '{0-role-name}' does not have 'authenticate' privilege on {1-service-name}             | |
+| 4  | Host is not defined in Conjur                                           | '{0-role-name}' wasn't found                                                           | |
+| 5  | Couldn't make connection with Google Identity Provider in time          | Google Identity Provider failed with timeout error (Provider URI: '{0}'). Reason: '{1}'| |
```

Comment:
@nessiLahav can we specify what is the timeout - BTW is it configurable somewhere? do we log the uri we called?

Reply:
Those message are aligned with Azure, at the development stage you will review them again
Yes we log the privder uri
the timeout is not configurable